front/src/app/todo-list/components/active-task-list/active-task-list.component.ts

```diff
@@ -0,0 +1,90 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { TaskList } from '../../../shared/models/shared.models';
+import { Observable, Subject, filter, switchMap, takeUntil } from 'rxjs';
+import { ApiService } from '../../../shared/api-service/api.service';
+
+@Component({
+  selector: 'app-active-task-list',
+  templateUrl: './active-task-list.component.html',
+  styleUrl: './active-task-list.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ActiveTaskListComponent implements OnDestroy, OnInit {
+  @Input() listId$: Observable<string> | undefined;
+
+  taskList: TaskList | undefined;
+
+  isNewTaskInputVisible: boolean = false;
+
+  activeListId: string = '';
+
+  private destroy$ = new Subject<void>();
+
+  constructor(public apiService: ApiService, private changeDetectorRef: ChangeDetectorRef) {
+  } 
+
+  ngOnInit(): void {
+    this.listId$?.pipe(
+      takeUntil(this.destroy$),

```

Comment:
As a rule `takeUntil` has to be the last operator in the pipe (otherwise some inner subscribtions like switchMap won't be canceled)